db/import_column_family_job.cc

```diff
@@ -61,12 +61,13 @@ Status ImportColumnFamilyJob::Prepare(uint64_t next_file_number,
                        info1->smallest_internal_key,
                        info2->smallest_internal_key) < 0;
           });
-
-      for (size_t i = 0; i < sorted_files.size() - 1; i++) {
-        if (cfd_->internal_comparator().Compare(
-                sorted_files[i]->largest_internal_key,
-                sorted_files[i + 1]->smallest_internal_key) >= 0) {
-          return Status::InvalidArgument("Files have overlapping ranges");
+      if (sorted_files.size() > 1) {
+        for (size_t i = 0; i < sorted_files.size() - 1; i++) {
```

Comment:
How about `i + 1 < sorted_files.size()` instead?

And it looks like there about about 24 more that would could apply this to pro-actively, `git grep -E 'for [(](size_t|uint|unsigned).* (<|!=) .*- 1;'`

Reply:
> And it looks like there about about 24 more that would could apply this to pro-actively

Feel free to leave this to a separate PR/task so that this one can land.

Reply:
@pdillinger Thanks for the review. Will change to i + 1 < sorted_files.size() in the for loop. Yeah, I will create separate PR for the other potential overflows you mentioned. Thanks!